app/concepts/reservations/use_cases/fetch_unpaid_reservations.rb

```diff
@@ -0,0 +1,18 @@
+module Reservations
+  module UseCases
+    class FetchUnpaidReservations
```

Comment:
I think that this class logic should be moved to your `Reservations::Repository` class. You can add for example method like this:
```
def unpaid
  adapter.where(status: "confirmed")
end
``` 
As @tobiaszwaszak mention, screenings shouldn't be important here.

Reply:
Ok, I saw that screening is important :) So you can have a method like this:
```
def unpaid(screening)
  adapter.where(status: "confirmed", screening_id: screening.id)
end
```

Reply:
Hmm I don't want to cancel all unpaid reservation, only those for given screening 30 minute before it starts. I will think how to reorganise it.